components/CredentialDetailsViews.vue

```diff
@@ -197,37 +199,28 @@ export default {
       backgroundColor: 'gray',
     };
 
-    // Fetch style, overrides, & highlights before component mounts
-    onBeforeMount(() => {
-      getDisplaysFromRenderMethod();
-    });
-
     // Extract and parse images from credential's render method property
     async function getDisplaysFromRenderMethod() {
       if(props.credential?.renderMethod?.length) {
-        props.credential.renderMethod.forEach(async rm => {
-          if(supportedRenderMethods.includes(rm.type)) {
-            if(rm.type === 'SvgRenderingTemplate2023') {
-              useRenderTemplate2023(rm.id);
-            } else if(rm.type === 'SvgRenderingTemplate2024') {
-              const {template, url} = rm;
-              const values = props.credential;
-              await useRenderTemplate2024(template, url, values);
+        const imageValues = await Promise.all(
+          props.credential.renderMethod.map(async rm => {
+            let imageValue;
+            if(supportedRenderMethods.includes(rm.type)) {
+              if(rm.type === 'SvgRenderingTemplate2023') {
+                imageValue = useRenderTemplate2023(rm.id);
+              } else if(rm.type === 'SvgRenderingTemplate2024') {
+                const {template, url} = rm;
+                const values = props.credential;
+                imageValue = await useRenderTemplate2024(template, url, values);
+              }
             }
-          }
-        });
+            return imageValue;
+          })
+        );
+        imageValues.filter(Boolean).forEach(v => credentialImages.push(v));
```

Comment:
A quick note that none of this is vue-specific code, so we should import a function from a vanilla JS lib and call that (probably from bedrock-web-wallet?). That other library should also be able to, with time use a proxy or use OHTTP to obtain these things -- and be able to interact with a cache to avoid retrieval more than once across components and so on.

Reply:
@dlongley 
In looking to refactor as described above, I noticed this function calls useRenderTemplate2024 that uses a UI library (Mustache) and a Quasar date library (date) to aid in the UI rendering logic.

Additionally, with renderMethod continuing to change in the future (before its become a stable standard) I thought it'd be best to leave this logic in this vue file for the time being.